pkg/archive/archive.go

```diff
@@ -530,7 +550,8 @@ func createTarFile(path, extractDir string, hdr *tar.Header, reader io.Reader, L
 		}
 
 	case tar.TypeLink:
-		targetPath := filepath.Join(extractDir, hdr.Linkname)
+		lpath := adjustHardLink(hdr.Linkname, hdr.Name)
+		targetPath := filepath.Join(extractDir, lpath)
```

Comment:
I agree that this is the wrong place to make this change. It should probably be done in the `RebaseArchiveEntries` function in the file `pkg/archive/copy.go`. There, it should check to see if `hdr.Typeflag == tar.TypeLink` and replace `hdr.Linkname` in the same way that it replaces `hdr.Name`:

```diff
--- a/pkg/archive/copy.go
+++ b/pkg/archive/copy.go
@@ -332,6 +332,9 @@ func RebaseArchiveEntries(srcContent io.Reader, oldBase, newBase string) io.Read
                        }
 
                        hdr.Name = strings.Replace(hdr.Name, oldBase, newBase, 1)
+                       if hdr.Typeflag == tar.TypeLink {
+                               hdr.Linkname = strings.Replace(hdr.Linkname, oldBase, newBase, 1)
+                       }
 
                        if err = rebasedTar.WriteHeader(hdr); err != nil {
                                w.CloseWithError(err)
```